File: src/main/java/pl/mroz/buddiesapi/domain/account/AccountDomainServiceImpl.java
package pl.mroz.buddiesapi.domain.account;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import pl.mroz.buddiesapi.domain.common.ObjectMapper;

import java.util.List;
import java.util.UUID;

@Slf4j
@RequiredArgsConstructor
public class AccountDomainServiceImpl implements AccountDomainService {

    private final AccountRepository repository;

    @Override
    public List<Account> getAllAccounts() {
        return repository.findAll();
    }

    @Override
    public Account getAccount(UUID accountId) {
        return repository.getById(accountId);
    }

    @Override
    public Account createAccount(Account account) {
        return repository.save(account);
    }

    @Override
    public Account updateAccount(UUID uuid, Account newAccountData) {
        var oldAccountData = getAccount(uuid);
        return repository.save(ObjectMapper.mapInto(newAccountData, oldAccountData));
    }

    @Override
    public void deleteAccount(Account account) {
        repository.delete(account);
    }
}

File: src/main/kotlin/buddiesapi/domain/account/AccountDomainServiceImpl.kt
package buddiesapi.domain.account

import buddiesapi.domain.common.ObjectMapper
import java.util.UUID
import mu.KLogging

class AccountDomainServiceImpl(val repository: AccountRepository) : AccountDomainService {

    override val allAccounts: List<Account> = repository.findAll()

    override fun getAccount(accountId: UUID): Account? =
        repository.getById(accountId)

    override fun createAccount(account: Account): Account {
        return repository.save(account)
    }

    override fun updateAccount(uuid: UUID, account: Account): Account =
        repository.save(ObjectMapper.mapInto(account, getAccount(uuid)!!))

    override fun deleteAccount(account: Account) =
        repository.delete(account)

    companion object : KLogging()
}
